tests/components/google_mail/test_init.py

```diff
@@ -0,0 +1,130 @@
+"""Tests for Google Mail."""
+import http
+import time
+from unittest.mock import patch
+
+from aiohttp.client_exceptions import ClientError
+import pytest
+
+from homeassistant.components.google_mail import DOMAIN
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr
+
+from .conftest import GOOGLE_TOKEN_URI, ComponentSetup
+
+from tests.test_util.aiohttp import AiohttpClientMocker
+
+
+async def test_setup_success(
+    hass: HomeAssistant, setup_integration: ComponentSetup
+) -> None:
+    """Test successful setup and unload."""
+    await setup_integration()
+
+    entries = hass.config_entries.async_entries(DOMAIN)
+    assert len(entries) == 1
+    assert entries[0].state is ConfigEntryState.LOADED
+
+    await hass.config_entries.async_unload(entries[0].entry_id)
+    await hass.async_block_till_done()
+
+    assert not len(hass.services.async_services().get(DOMAIN, {}))
```

Comment:
We don't need to check the length of the container. Containers are truthy if they are not empty and falsy if they are empty.